.github/actions/gh-pages-storybook/index.js

```diff
@@ -0,0 +1,44 @@
+const core = require('@actions/core');
+const exec = require('@actions/exec');
+
+async function run () {
+  try {
+    if (core.isDebug()) {
+      core.startGroup('Debug: context');
+      core.debug(core.context);
+      core.endGroup();
+    }
+
+    const buildPath = 'storybook-static';
+    
+    core.info('[INFO]: commencing storybook gh-pages deploy')
+
+    core.info('[INFO]: running storybook build')
+    await exec.exec('npm', ['run', 'build-storybook']);
+
+    core.info('[INFO]: configuring git')
+    await exec.exec('git', ['config', '--global', 'user.email', 'github-actions[bot]@users.noreply.github.com']);
+    await exec.exec('git', ['config', '--global', 'user.name', 'github-actions[bot]']);
+
+    core.info('[INFO]: checking out gh-pages branch')
+    // -B: don't error if the branch already exists
+    await exec.exec('git', ['checkout', '-B', 'gh-pages']);
+
+    core.info('[INFO]: adding storybook static files')
+    // --force: override .gitignore
+    await exec.exec('git', ['add', '--force', buildPath]);
+
+    core.info('[INFO]: committing changes');
+    // TODO: retrieve latest release and put it in the commit msg, token does not have privileges
```

Comment:
You mean the access token doesn't have privileges?

Reply:
as opposed to `does not`? not sure why I used long hand version.

Reply:
no, sorry :D I meant to ask if you maybe only checked with the `GIHUB_TOKEN` instead of the `ACCESS_TOKEN` ;)

Reply:
yep, I tried with [@actions/github](https://www.npmjs.com/package/) which is already hydrated with the GITHUB_TOKEN. I'm not sure we have another token? we could create one, but I'd rather the pipeline not having too many privileges. What do you think? there may be a more fine grained option, I was running a bit low on time.